src/handlers/psp.ts

```diff
@@ -0,0 +1,436 @@
+        return;
+      }
+
+      // Get endpoint details
+      const endpoint = config.endpoints[template.endpointId];
+      if (isNil(endpoint)) {
+        node.logger.warn(`EndpointId ${template.endpointId} not found in endpoints`, baseLogOptions);
+        return;
+      }
+      // Verify endpointId
+      const expectedEndpointId = ethers.utils.keccak256(
+        ethers.utils.defaultAbiCoder.encode(['string', 'string'], [endpoint.oisTitle, endpoint.endpointName])
+      );
+      if (expectedEndpointId !== template.endpointId) {
+        node.logger.warn(
+          `EndpointId ${template.endpointId} does not match expected ${expectedEndpointId}`,
+          baseLogOptions
+        );
+        return;
+      }
+
+      groupedSubscriptions.push({
+        subscriptions: enabledSubscriptionsByTemplateId[templateId],
+        template: { ...template, id: templateId },
+        endpoint: { ...endpoint, id: template.endpointId },
+      });
+    });
+  }
+
+  return {
+    config,
+    baseLogOptions,
+    groupedSubscriptions,
+    apiValuesBySubscriptionId: {},
+    providerStates: [],
+  };
+};
+
+const executeApiCalls = async (state: State): Promise<State> => {
+  const { config, baseLogOptions, groupedSubscriptions } = state;
+
+  // TODO: promise.all? 🤔
+  let apiValuesBySubscriptionId: { [subscriptionId: string]: ethers.BigNumber } = {};
+  for (const { subscriptions, template, endpoint } of groupedSubscriptions) {
+    const templateLogOptions: node.LogOptions = {
+      ...baseLogOptions,
+      additional: {
+        templateId: template.id,
+      },
+    };
+    const apiCallParameters = abi.decode(template.templateParameters);
+    const [errorCallApi, logsData] = await retryGo(() =>
+      callApi({
+        oises: config.ois,
+        apiCredentials: config.apiCredentials,
+        apiCallParameters,
+        oisTitle: endpoint.oisTitle,
+        endpointName: endpoint.endpointName,
+      })
+    );
+    if (!isNil(errorCallApi) || isNil(logsData)) {
+      node.logger.warn('Failed to fecth API value', templateLogOptions);
+      continue;
+    }
+    const [logs, apiValue] = logsData;
+    node.logger.logPending(logs, templateLogOptions);
+
+    if (isNil(apiValue)) {
+      node.logger.warn('Failed to fetch API value. Skipping update...', templateLogOptions);
+      continue;
+    }
+
+    apiValuesBySubscriptionId = {
+      ...apiValuesBySubscriptionId,
+      ...subscriptions.reduce((acc, subscription) => ({ ...acc, [subscription.id]: apiValue }), {}),
+    };
+  }
+
+  return { ...state, apiValuesBySubscriptionId };
+};
+
+const initializeProviders = async (state: State): Promise<State> => {
+  const { config, baseLogOptions } = state;
+
+  const airnodeWallet = ethers.Wallet.fromMnemonic(config.nodeSettings.airnodeWalletMnemonic);
+
+  const evmChains = config.chains.filter((chain: ChainConfig) => chain.type === 'evm');
```

Comment:
Just fyi you don't need the type:
```
const evmChains = config.chains.filter(chain => chain.type === 'evm');
```

Reply:
I guess I did it because airnode-node is also [checking](https://github.com/api3dao/airnode/blob/master/packages/airnode-node/src/providers/actions.ts#L47) the chain type but yes. It's probably not necessary.

Reply:
No no, I meant you don't need `(chain: ChainConfig)` only `chain` ^
I can't comment on the actual check, but I'd imagine it is indeed unnecessary.